cpp/src/quantiles/quantiles.cu

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ #pragma once
+
+ #include <cudf/table/table.hpp>
+ #include <cudf/table/table_view.hpp>
+ #include <cudf/quantiles.hpp>
+ #include <cudf/scalar/scalar.hpp>
+ #include <cudf/scalar/scalar_factories.hpp>
+ 
+ namespace cudf {
+     
+ namespace experimental {
+ 
+ /* @brief Computes the quantile of any sorted arithmetic column.
+  *
+  * @param[in] in                     Column from which quantile is computed.
+  * @param[in] quantile_interpolation Strategy to obtain a quantile which falls
+                                      between two points.
+  *
+  * @returns The quantile within range [0, 1]
+  */
+ std::unique_ptr<scalar> quantile(column_view const& in,
+                                  double quantile,
```

Comment:
I think the API is "quantiles", not "quantile", and it takes a vector of quantiles to compute and returns multiple values. Is this not the case?

Reply:
It should be `quantiles`. The API you’re thinking of is `group_quantiles`. I have not added that yet, as I am not sure if it belongs in this PR or a PR related to groupby.

Reply:
Ah, I think the intent is that eventually we should be able to take a vector of quantiles to return, and return multiple, as in Pandas (https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.quantile.html).  But I see that was never implemented, so please continue just porting what we had. 

group_quantiles is indeed related to groupby, please discuss with @devavret whether to port as part of this PR or if it will be ported as part of groupby.

Reply:
No need to port group quantiles. That’ll be a part of groupby.

Reply:
@harrism Should we keep the name `quantile` (singular), seeing as how it only returns a single quantile? Or, for the sake of _future_ consistency, keep the name `quantiles` (plural) to account for the multiple-quantile return function you mentioned?

Reply:
You know you could just add the functionality to do multiple quantiles right now. And since the expected size of q vector isn’t too large, all you need to do is put a for loop onto the existing logic.